### ISSUE TITLE
'apply.daterangepicker' event listener not triggered when user updates text in input field via keyboard.

### ISSUE THREAD

**Issue by craigdanj** [dangrossman/bootstrap-daterangepicker#1710]
I noticed that the 'apply.daterangepicker' event handler isn't called when text is typed into the input field thus requiring the creation of a separate function just to handle that particular scenario. 

I think it would be helpful if the plugin itself handled this. Would this be possible?

**Comment by weisborg** [comment]
Yeah, this was a hard bug to figure out in my group too.  We had assumed it was bound to the input and handled it all, but it doesn't.  In our case we have two hidden fields to hold the From and To dates.  On Apply we grab startDate and endDate and set those fields.  So, when users typed values and never used the picker nothing was saved.  I ended up using an on input event for the input and split the values, validated the dates, and copied to the hidden if valid or set them to empty if not.

We also have issues with the time keeping in sync.  Dropdowns match what they were last picked or midnight and not what is in the field, etc.  autoUpdateInput being set to true is the only thing that seems tied to keeping things in sync like the time dropdowns, but it also sets a default value to current datetime. Edit pages where values are set at load have issues too where the control will set current datetime, but when clicked it reverts back to the correct time entered.  This really messes up things where users go in to edit something else and all the date picker fields they didn't touch get saved to current datetime.
